moveit_ros/manipulation/pick_place/src/manipulation_pipeline.cpp

```diff
@@ -110,30 +110,30 @@ void ManipulationPipeline::start()
 {
   stop_processing_ = false;
   empty_queue_threads_ = 0;
-  for (std::size_t i = 0; i < stages_.size(); ++i)
-    stages_[i]->resetStopSignal();
+  for (pick_place::ManipulationStagePtr& stage : stages_)
+    stage->resetStopSignal();
   for (std::size_t i = 0; i < processing_threads_.size(); ++i)
     if (!processing_threads_[i])
       processing_threads_[i] = new boost::thread(boost::bind(&ManipulationPipeline::processingThread, this, i));
 }
 
 void ManipulationPipeline::signalStop()
 {
-  for (std::size_t i = 0; i < stages_.size(); ++i)
-    stages_[i]->signalStop();
+  for (pick_place::ManipulationStagePtr& stage : stages_)
+    stage->signalStop();
   stop_processing_ = true;
   queue_access_cond_.notify_all();
 }
 
 void ManipulationPipeline::stop()
 {
   signalStop();
-  for (std::size_t i = 0; i < processing_threads_.size(); ++i)
-    if (processing_threads_[i])
+  for (boost::thread* processing_thread : processing_threads_)
+    if (processing_thread)
     {
-      processing_threads_[i]->join();
-      delete processing_threads_[i];
-      processing_threads_[i] = nullptr;
```

Comment:
This is the first time I've seen this, but by iterating over pointers, `processing_thread = nullptr;`doesn't correctly change the entry in the vector, it's only changing the copy of the pointer. Should probably just use `//NOLINT` here. 

I tried to look for other instances of this, but this is the main way where bugs can be introduced I believe.

Reply:
True, but I think maybe we don't need use `//NOLINT`, it would be enough if changing the range-declaration from pointer to pointer reference for this kind of cases.

Reply:
That's true, pointers references would work too.